src/codegen/compile.ml

```diff
@@ -3980,6 +4095,113 @@ module StableMem = struct
           get_n32 ^^
           write_word32 env)
     | _ -> assert false
+
+  let load_word8 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code1 env "__stablemem_load_word8"
+        (("offset", I32Type)) [I32Type]
+        (fun env get_offset  ->
+          get_offset ^^
+          guard env ^^
+          get_offset ^^
+          read_word8 env)
+    | _ -> assert false
+
+  let store_word8 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code2 env "__stablemem_store_word8"
+        (("offset", I32Type), ("value", I32Type)) []
+        (fun env get_offset get_value ->
+          get_offset ^^
+          guard env ^^
+          get_offset ^^
+          get_value ^^
+          write_word8 env)
+    | _ -> assert false
+
+  let load_word16 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code1 env "__stablemem_load_word16"
+        (("offset", I32Type)) [I32Type]
+        (fun env get_offset  ->
+          get_offset ^^
+          compile_unboxed_const 2l ^^
+          guard_range env ^^
+          get_offset ^^
+          read_word16 env)
+    | _ -> assert false
+
+  let store_word16 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code2 env "__stablemem_store_word16"
+        (("offset", I32Type), ("value", I32Type)) []
+        (fun env get_offset get_value ->
+          get_offset ^^
+          compile_unboxed_const 2l ^^
+          guard_range env ^^
+          get_offset ^^
+          get_value ^^
+          write_word16 env)
+    | _ -> assert false
+
+  let load_word64 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code1 env "__stablemem_load_word64"
+        (("offset", I32Type)) [I64Type]
+        (fun env get_offset  ->
+          get_offset ^^
+          compile_unboxed_const 8l ^^
+          guard_range env ^^
+          get_offset ^^
+          read_word64 env)
+    | _ -> assert false
+
+  let store_word64 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code2 env "__stablemem_store_word64"
+        (("offset", I32Type), ("value", I64Type)) []
+        (fun env get_offset get_value ->
+          get_offset ^^
+          compile_unboxed_const 8l ^^
+          guard_range env ^^
+          get_offset ^^
+          get_value ^^
+          write_word64 env)
+    | _ -> assert false
+
+  let load_float64 env =
+    match E.mode env with
+    | Flags.ICMode | Flags.RefMode ->
+      Func.share_code1 env "__stablemem_load_float64"
+        (("offset", I32Type)) [F64Type]
+        (fun env get_offset  ->
+          get_offset ^^
+          compile_unboxed_const 8l ^^
+          guard_range env ^^
+          get_offset ^^
+          read_float64 env)
+    | _ -> assert false
+
+  let store_float64 env =
```

Comment:
Around line 3936 there is a similar function? What is the semantic difference between `__stablemem_store_float64` and `__stablemem_write_float64`?

Reply:
I presume the `load`/`store` ones operate on guarded ranges?